src/libsync/owncloudpropagator.cpp

```diff
@@ -469,7 +469,9 @@ void OwncloudPropagator::start(SyncFileItemSet &&items)
             if (item->_instruction == CSYNC_INSTRUCTION_REMOVE) {
                 // We do the removal of directories at the end, because there might be moves from
                 // these directories that will happen later.
-                directoriesToRemove.prepend(dir);
+                Q_ASSERT(!removeDirectoryJob);
```

Comment:
It appears that we might set removeDirectoryJob  multiple times, I assume it is a clever part of an algorithm.
removeDirectoryJob  is accessed in 402